.azure-pipelines/run-test-elastictest-template.yml

```diff
@@ -0,0 +1,251 @@
+parameters:
+  - name: TOPOLOGY
+    type: string
+
+  - name: POLL_INTERVAL
+    type: number
+    default: 10
+
+  - name: POLL_TIMEOUT
+    type: number
+    default: 36000
+
+  - name: MIN_WORKER
+    type: string
+    default: 1
+
+  - name: MAX_WORKER
+    type: string
+    default: 1
+
+  - name: NUM_ASIC
+    type: number
+    default: 1
+
+  - name: TEST_SET
+    type: string
+    default: ""
+
+  - name: DEPLOY_MG_EXTRA_PARAMS
+    type: string
+    default: ""
+
+  - name: COMMON_EXTRA_PARAMS
+    type: string
+    default: ""
+
+  - name: VM_TYPE
+    type: string
+    default: "ceos"
+
+  - name: TESTBED_NAME
+    type: string
+    default: ""
+
+  - name: IMAGE_URL
+    type: string
+    default: ""
+
+  - name: HWSKU
+    type: string
+    default: ""
+
+  - name: TEST_PLAN_TYPE
+    type: string
+    default: ""
+
+  - name: PLATFORM
+    type: string
+    default: ""
+
+  - name: SCRIPTS
+    type: string
+    default: ""
+
+  - name: FEATURES
+    type: string
+    default: ""
+
+  - name: SCRIPTS_EXCLUDE
+    type: string
+    default: ""
+
+  - name: FEATURES_EXCLUDE
+    type: string
+    default: ""
+
+  - name: REPO_NAME
+    type: string
+    default: ""
+
+  - name: MGMT_BRANCH
+    type: string
+    default: ""
+
+  - name: MGMT_URL
+    type: string
+    default: "https://raw.githubusercontent.com/sonic-net/sonic-mgmt"
+
+  - name: STOP_ON_FAILURE
+    type: string
+    default: ""
+
+  - name: RETRY_TIMES
+    type: string
+    default: ""
+
+  - name: DUMP_KVM_IF_FAIL
+    type: string
+    default: "True"
+    values:
+      - "True"
+      - "False"
+
+  - name: REQUESTER
+    type: string
+    default: ""
+
+  - name: MAX_RUN_TEST_MINUTES
+    type: number
+    default: 480
+
+  - name: KVM_IMAGE_BRANCH
+    type: string
+    default: ""
+
+  - name: DOWNLOAD_TEST_PLAN_SCRIPTS
+    type: boolean
+    default: false
+
+  - name: RUN_TEST_TO_THE_END
+    type: boolean
+    default: false
+
+
+steps:
+  - ${{ if eq(parameters.DOWNLOAD_TEST_PLAN_SCRIPTS, true) }}:
+      - script: |
+          set -ex
+
+          curl "https://raw.githubusercontent.com/sonic-net/sonic-mgmt/master/.azure-pipelines/test_plan.py" -o ./.azure-pipelines/test_plan.py
+
+          if [[ -z "$(AZP_REPO_ACCESS_TOKEN)" ]]; then
```

Comment:
need to separate another step for the pr_test_scripts.yaml.
if repo is sonic-mgmt
     not download pr_test_scripts.yaml, otherwise download pr_test_scripts.yaml
     if branch is master: will not download test_plan,py, otherwise download test_plan.py